templates/hooks/endpoint/sdk_update_pre_set_output.go.tpl

```diff
@@ -0,0 +1,6 @@
+
+// always requeue with desired state and return immediately due to eventually
+// consistent API
+return desired, ackrequeue.NeededAfter(nil, defaultRequeueDelay)
+
+// TODO (@embano1): we can't tell code-gen to not generate the rest of the code
```

Comment:
You could set:

```yaml
resources:
  Endpoint:
    reconcile:
      requeue_on_success_seconds: 60
```

In your `generator.yaml` file to handle this... See here:

https://aws-controllers-k8s.github.io/community/docs/contributor-docs/code-generator-config/#reconcile-controlling-reconciliation-and-requeue-logic

Reply:
IIUC, our logic is a bit more complex:

- in `create`, only requeue if `!availableState` with **specific** message within 5s
- in `update`, always requeue with **specific** message within 5s (IIUC, in case of failure the controller uses a different [longer] requeue delay)

I guess the different delays (`requeue_on_success_seconds` and hardcoded `err` requeue in the controller) as well as the required specific message we set on the resource requires custom logic here?

Reply:
I think that `requeue_on_success_seconds` will not solve this problem. It's more of an API consistency issue... (API will not return that a resource is UPDATING, until a few seconds later.. which will bait `requeue_on_success_seconds`) (correct me if i'm wrong)

> always requeue with a specific message within 5s (IIUC, in case of failure the controller uses a different [longer] requeue delay)

Exponential backoff requeues will only happen if we don't hardcode a requeue value